pkg/manifest/ostree_deployment.go

```diff
@@ -340,7 +340,8 @@ func (p *OSTreeDeployment) serialize() osbuild.Pipeline {
 		}
 	}
 
-	if !hasRoot {
+	if !hasRoot && p.ostreeSpec != nil {
+		// NOTE: fails on container-based deployments
```

Comment:
not sure either.. from my side this feels like something that would be done when composing the ostree commit or oscontainer rather than when composing an image. IOW I feel like this shouldn't be part of Image Builder at all?

Reply:
Locking the root password during deployment is something that's done for the official Fedora IoT image: https://pagure.io/fedora-kickstarts/blob/main/f/fedora-iot.ks#_9

We were replicating that here.